src/LiveComponent/src/Test/InteractsWithLiveComponents.php

```diff
@@ -44,4 +44,24 @@ protected function createLiveComponent(string $name, array $data = [], ?KernelBr
             self::getContainer()->get('router'),
         );
     }
+
+    protected function assertComponentEmitEvent(TestLiveComponent $testLiveComponent, string $eventName, ?array $parameters = null): void
+    {
+        $eventData = $testLiveComponent->getEmittedEvent($testLiveComponent->render(), $eventName);
+
+        $this->assertNotNull($eventData, \sprintf('Event %s not emitted', $eventName));
+
+        if (null === $parameters) {
+            return;
+        }
+
+        foreach ($parameters as $key => $value) {
+            $this->assertSame($value, $eventData['data'][$key] ?? null, \sprintf('EventData (%s) is not valid', $key));
+        }
```

Comment:
It will helps by displaying a nice diff between expected vs actual data:
```suggestion
        $this->assertEquals(expectedEventData, $event['data'], sprintf('The expected event  "%s" data does not match.', $eventName));
```

Reply:
Yep but with the global assertEquals, we must pass all expectedEventDatas.

If the event contains ['a' => 'b', 'c' => 'd'];
And I expect only ['a' => 'b']; 
It throw an error.

Reply:
That's true, but I think that's the desired behavior. Asserting the whole event data makes more sense to me, but I may be biaised and I understand that sometimes you expect to assert an array subset, but it must be explicit. 

I've seen it tons of time on multiple projects, where sometimes you assert everything, or sometimes only a subset, and every time that's a mess because the method-name / PHPUnit's way of working does not help.

I can suggest you this, so everyone would be happy:
```php
// classic way
$this->assertComponentEmitEvent($component, 'foo'); // do not care about event data
$this->assertComponentEmitEventWithData($component, 'foo', []); // assert whole array
$this->assertComponentEmitEventWithDataSubset($component, 'foo', []); // assert subset array

// or a bit exotic 
$this->assertComponentEmitEvent($component, 'foo'); // do not care about event data
$this->assertComponentEmitEvent($component, 'foo')->withData([]); // assert whole array
$this->assertComponentEmitEvent($component, 'foo')->withDataSubset([]); // assert subset array
```

WDYT?

Reply:
> // or a bit exotic 
> $this->assertComponentEmitEvent($component, 'foo'); // do not care about event data
> $this->assertComponentEmitEvent($component, 'foo')->withData([]); // assert whole array
> $this->assertComponentEmitEvent($component, 'foo')->withDataSubset([]); // assert subset array

I personally **_LOVE_** this! 😍

Reply:
I tested the `->with` implementation

Reply:
An anonym class is enough or I have to separate it ?

The limit of the anonyme class is the dot block : I can not chained the `->with`

Reply:
A separate concrete class please :)